### PR TITLE
Handbrake, failed mat/demat effects and many other changes

### PR DESCRIPTION
**Pull request by parar020100**
Includes: #469, it should be merged first
Related issues: #357 #456 #446 #368 #337 #403

- Handbrake control, prevents TARDIS from taking off
  - Sounds for failed demat
  - HADS makes an exception
  - Handbrake while demating cancells teleport with damage, sparks, explosions, power off
    - A setting to remove those effects for weak computers
    - Power can't be turned back on in 10 seconds after crash landing
  - Handbrake at vortex or while materialising results in **same**, but also with falling out at a random location in the map  
- Made impossible to demat with doors open
  - HADS makes an exception
  - Added a setting to disable that
- Engine release, allows demat with opened doors or handbrake with broken effects (shaking, explosions, sparks)
  - Only works if pressed right after failed demat (several seconds)
  - A setting to remove those effects for weak computers  
- Destination select improvements
  - Increased destination speed boost with "shift" 
  - "H" key in third person opens destination select
  - **Random location select in coordinates menu** (2 options - just random and random on the ground)
- Sounds for damaged TARDIS teleportation
- Fixed poweroff being possible in the vortex 
- Disabled cloister bells if power is disabled
- Fixed message for float in third person
- [internal] Fixed removing parts in interior overrides (Parts = { ... <part> = false, ... }

### CODE REVIEW

**Comment by MattJeanes** [comment]
Haven't reviewed the code yet but playing around with it is really fun! Loving the new sounds and effects! I got a little confused at first why it wouldn't take off with the doors open, I think perhaps we should either add a message prompt or make it close them by default as it did before.

Some observations from testing:
- Power can get stuck at 0% by blowing it up in the vortex from the inside
- The failed demat sound gets cut off at the end
- Force materialising while in the vortex with the doors open results in not being able to walk through the doors until you close and re-open them again
- The explosion effects are visible for every player across the entire map, especially the sparks as they rain down from the sky - maybe set a much shorter TTL on these if we can?
image

File: lua/entities/gmod_tardis/modules/sh_handbrake.lua
Comment (MattJeanes) [inline lua/entities/gmod_tardis/modules/sh_handbrake.lua:16]: This needs an icon

File: lua/entities/gmod_tardis_interior/modules/sh_cloisterbells.lua
Comment (MattJeanes) [inline lua/entities/gmod_tardis_interior/modules/sh_cloisterbells.lua:25]: This should use GetPower()

**Review by MattJeanes — CHANGES_REQUESTED** [review]

File: lua/entities/gmod_tardis/modules/sh_lock.lua
Comment (MattJeanes) [inline lua/entities/gmod_tardis/modules/sh_lock.lua:23]: This control appears to have the same order as the door open - not sure if intentional

File: lua/tardis/themes/visgui/default-transparent.lua
Comment (MattJeanes) [inline lua/tardis/themes/visgui/default-transparent.lua:32]: Probably shouldn't be a new line here breaking the two sections up, same below as well

File: lua/entities/gmod_tardis/modules/sh_lock.lua
Comment (parar020100) [reply lua/entities/gmod_tardis/modules/sh_lock.lua:23]: It is intentional. The order of those doesn't matter, they are technically in the same category. What matters is that they're next to each other and after some certain ones. The sorting supports equal order numbers, I've checked that.

File: lua/tardis/themes/visgui/default-transparent.lua
Comment (parar020100) [reply lua/tardis/themes/visgui/default-transparent.lua:32]: Fixed

**Comment by parar020100** [comment]
> The explosion effects are visible for every player across the entire map
They are only seen by TARDIS occupants, that is checked. And disabling it in third person would be complicated, since we'd have to check for third person every time we do those effects, it would slow it down, and, imho, not really worth the effort.

> Power can get stuck at 0% by blowing it up in the vortex from the inside
Fixed, now blowing up in the vortex results in falling out of it

> Force materialising while in the vortex with the doors open results in not being able to walk through the doors until you close and re-open them again
Fixed, interrupting the teleport now closes the doors automatically

> The failed demat sound gets cut off at the end
- Couldn't reproduce, it doesn't for me.
-

**Comment by MattJeanes** [comment]
It is specifically the demat_fail_ext.wav that gets cut off, the interior variant is fine.

Blowing up in the vortex already resulted in falling out of it but the health then got stuck at 0% and it was not possible to kill the TARDIS completely afterwards by shooting it more it would just stay in the warning state - is this what you have fixed?

Are you sure it is only visible to occupants? That effect would be visible globally as it's created serverside as part of the explosion - I would recommend using util.Effect / explosion sounds clientside instead and perhaps taking damage directly rather than creating an explosion on the interior

**Review by MattJeanes — APPROVED** [review]
I'm happy with these changes, definitely is going to need some testing as these are some core changes to the teleport systems but it's looking pretty good from having a play around. As mentioned I'll await any feedback from you before merging this in but I'm happy to merge if you are.

**Pull request merged**
